### PR TITLE
tests/main/selinux-lxd: make sure LXD from snaps works cleanly with enforcing SELinux

### PR DESCRIPTION
**Pull request by bboozzoo**
Add a test that checks whether LXD from snaps works with SELinux in enforcing mode and raises no denials related to snapd policy.

Since we do not have a sysctl backend, add a workaround for a known problem with LXD on CentOS.

This should be the final PR in the SELinux series.

(cc @Conan-Kudo )

### CODE REVIEW

File: data/sysctl/rhel7-snap.conf [hunk 1]
```diff
@@ -2,3 +2,10 @@
 # Unexpected "Device or resource busy" error when removing a directory
 # see https://access.redhat.com/articles/3128691 for details
 fs.may_detach_mounts=1
+
+# RHEL 7.4+ (and LXD) specific:
+# A configuration change was made in the 7.4 release, the default number of user
+# namespaces is set to zero, see https://access.redhat.com/solutions/3188102 for
+# details. For details about LXD side, see
+# https://discuss.linuxcontainers.org/t/lxd-on-centos-7/1250
+user.max_user_namespaces=15000
```
Comment (stolowski) [inline data/sysctl/rhel7-snap.conf]: It's a bit unfortunate we have to touch a global limit, but well... 
The forum topic you linked seems to be suggesting much lower value, any reason to go higher?

File: tests/main/selinux-lxd/task.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+description: |
+    Make sure that LXD snap can be installed and used on a SELinux enable
+    system, with enforcing mode on.
+
+# Systems with SELinux enabled out of the box.
+systems: [fedora-*, centos-*]
+
+# lxd downloads can be quite slow
+kill-timeout: 25m
```
Comment (stolowski) [inline tests/main/selinux-lxd/task.yaml]: Do we really need this (to be that large)? FWTW tests/regression/lp-1815869 installs lxd snap with no kill-timeout defined.

File: tests/main/selinux-lxd/task.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+# Start before anything else as it can take a really long time.
+priority: 1000
```
Comment (stolowski) [inline tests/main/selinux-lxd/task.yaml]: Why do we want to start it with higher prio?

File: data/sysctl/rhel7-snap.conf (hunk 1 above)
Comment (bboozzoo) [reply data/sysctl/rhel7-snap.conf]: That's the value suggested by RHEL knowledge base solution.

File: tests/main/selinux-lxd/task.yaml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    lxd.lxc launch "ubuntu:18.04" my-ubuntu
+
+    echo "Ensure we can run things inside"
+    lxd.lxc exec my-ubuntu echo hello | MATCH hello
+
+    echo "Stop the container"
+    lxd.lxc stop --force my-ubuntu
+
+    snap remove lxd
```
Comment (stolowski) [inline tests/main/selinux-lxd/task.yaml]: Should we do this in restore, and also remove the container? (see the existing lxd test I mentioned earlier).

File: tests/main/selinux-lxd/task.yaml (hunk 2 above)
Comment (bboozzoo) [reply tests/main/selinux-lxd/task.yaml]: Took the timeout and priority values from here: https://github.com/snapcore/snapd/blob/master/tests/main/lxd/task.yaml#L12-L16

File: tests/main/selinux-lxd/task.yaml (hunk 4 above)
Comment (bboozzoo) [reply tests/main/selinux-lxd/task.yaml]: Interstingly, the `tests/main/lxd` test does not remove the containers it creates. Perhaps what the regression test does is unnecessary?

Comment (stolowski) [reply tests/main/selinux-lxd/task.yaml]: IMHO we should remove the containers not to have leftovers that could affect other tests. At the moment they are probably wiped from the disk on lxd snap removal, but with the automatic snapshots we will be needlessly creating a snapshot here.

File: tests/main/selinux-lxd/task.yaml (hunk 2 above)
Comment (stolowski) [reply tests/main/selinux-lxd/task.yaml]: Hmmm, ok.

File: tests/main/selinux-lxd/task.yaml (hunk 3 above)
Comment (bboozzoo) [reply tests/main/selinux-lxd/task.yaml]: Sam motivation as here https://github.com/snapcore/snapd/pull/6711#discussion_r274840547

File: tests/main/selinux-lxd/task.yaml (hunk 4 above)
Comment (bboozzoo) [reply tests/main/selinux-lxd/task.yaml]: Pushed an update

Comment (stolowski) [reply tests/main/selinux-lxd/task.yaml]: Thanks!

**Pull request merged**